File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        if n==1:
            return '0'
        len=(1<<n)
        if(k==len//2):
            return '1'
        if(k<len//2):
            return self.findKthBit(n-1,k)
        return '1' if self.findKthBit(n-1,len-k)=='0' else '0'
        

File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    string inv(string s)
    {
        string res="";
        for(auto i:s)
        if(i=='0')
        res+='1';
        else
        res+='0';
        return res;
    }
    char findKthBit(int n, int k) {
        if(n==1)
        return '0';
        string s1="0";
        for(int i=1;i<n;i++)
        {
            string temp=s1;
            s1+='1';
            reverse(temp.begin(),temp.end());
            s1+=inv(temp);
        }
        return s1[k-1];
    }
};